projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectInformationService.java

```diff
@@ -98,35 +95,15 @@ public void addSpeciesToActiveExperiment(String projectId, Species... species) {
       return;
     }
     Arrays.stream(species).forEach(Objects::requireNonNull);
-
     ProjectId id = ProjectId.parse(projectId);
-    Optional<Project> optionalProject = projectRepository.find(id);
-
-    optionalProject.ifPresentOrElse(
-        project -> loadActiveExperimentForProject(project)
-            .ifPresentOrElse(addSpeciesToExperimentAndSave(species),
-                () -> addExperimentToProjectService
-                    .addExperimentToProject(project.getId(),
-                        "Experiment 0",
-                        List.of(),
-                        List.of(species),
-                        List.of())),
-        () -> {
-          throw new ProjectManagementException("There is no project with id " + id.value());
-        }
-    );
-  }
-
-  private Consumer<Experiment> addSpeciesToExperimentAndSave(Species[] species) {
-    return activeExperiment -> {
-      activeExperiment.addSpecies(List.of(species));
-      experimentRepository.update(activeExperiment);
-    };
+    Project project = loadProject(id);
+    Experiment activeExperiment = loadActiveExperimentForProject(project);
+    activeExperiment.addSpecies(List.of(species));
+    experimentRepository.update(activeExperiment);
```

Comment:
same with these: is that logically part of the ProjectInformationService? Shouldn't there be an ExperimentService?

Reply:
depends on what you consider the aggregate to be. Interactions should go over the aggregate. We could introduce an additional experiment information service or even smaller independent application services only responsible for information retrieval.